### DISCUSSION TITLE
Jupyter notebook install via poetry does not work, install via pip inside poetry created env works

### DISCUSSION THREAD

**Discussion by stephsi** [python-poetry/poetry#9289] [General] [answered]
When I try to install [Jupyter Notebook](https://pypi.org/project/notebook/) using

`poetry add notebook --group jupyter`
and 
`poetry install --with jupyter`

the installation runs through without any issues. 

But trying to start jupyter using `poetry run jupyter notebook / poetry run jupyter lab`

results in a ModuleNotFoundError

`ModuleNotFoundError: No module named 'fastjsonschema'`

If I now activate the virtual env created by poetry and run `pip install notebook`
it recognizes many requirements are already satisfied but installs all the dependencies missed by poetry like
```
Collecting babel>=2.10 (from jupyterlab-server<3,>=2.22.1->notebook)
  Using cached Babel-2.14.0-py3-none-any.whl.metadata (1.6 kB)
Collecting json5>=0.9.0 (from jupyterlab-server<3,>=2.22.1->notebook)
  Using cached json5-0.9.24-py3-none-any.whl.metadata (30 kB)
Collecting jsonschema>=4.18.0 (from jupyterlab-server<3,>=2.22.1->notebook)
  Using cached jsonschema-4.21.1-py3-none-any.whl.metadata (7.8 kB)
```

After the pip installation, deactivating the env and running `poetry run jupyter notebook / poetry run jupyter lab` works as expected now.
Manually adding all missing modules to the specific poetry dependency group does not seem like a solution to me.

Because the pip install inside the virtual env worked, I think this problem is related to poetry and thus I opened this ticket here.

I am using Windows 10, poetry version 1.7.1 installed via pipx.

Thank you very much in advance.

**Comment by dimbleby** [comment, marked as answer]
probably duplicate https://github.com/python-poetry/poetry/issues/9191, https://github.com/python-poetry/poetry/issues/9195, https://github.com/python-poetry/poetry/issues/9244, https://github.com/python-poetry/poetry/issues/9266, https://github.com/python-poetry/poetry/issues/9272, #9288 - with same solution.

**Comment by stephsi** [comment]
Thanks for the answer.
Updating poetry and clearing the caches resolved the issue. Can be closed.
